pkg/controller/keyvault/placeholder_pod_test.go

```diff
@@ -161,6 +181,115 @@ func TestPlaceholderPodControllerIntegration(t *testing.T) {
+		Replicas:             &replicas,
+		RevisionHistoryLimit: &historyLimit,
+		Selector:             &metav1.LabelSelector{MatchLabels: expectedLabels},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: expectedLabels,
+				Annotations: map[string]string{
+					"kubernetes.azure.com/observed-generation": "123",
+					"kubernetes.azure.com/purpose":             "hold CSI mount to enable keyvault-to-k8s secret mirroring",
+					"kubernetes.azure.com/ingress-owner":       ing.Name,
+					"openservicemesh.io/sidecar-injection":     "disabled",
+				},
+			},
+			Spec: *manifests.WithPreferSystemNodes(&corev1.PodSpec{
+				AutomountServiceAccountToken: util.BoolPtr(false),
+				Containers: []corev1.Container{{
+					Name:  "placeholder",
+					Image: "test-registry/oss/kubernetes/pause:3.6-hotfix.20220114",
+					VolumeMounts: []corev1.VolumeMount{{
+						Name:      "secrets",
+						MountPath: "/mnt/secrets",
+						ReadOnly:  true,
+					}},
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("20m"),
+							corev1.ResourceMemory: resource.MustParse("24Mi"),
+						},
+					},
+				}},
+				Volumes: []corev1.Volume{{
+					Name: "secrets",
+					VolumeSource: corev1.VolumeSource{
+						CSI: &corev1.CSIVolumeSource{
+							Driver:           "secrets-store.csi.k8s.io",
+							ReadOnly:         util.BoolPtr(true),
+							VolumeAttributes: map[string]string{"secretProviderClass": spc.Name},
+						},
+					},
+				}},
+			}),
+		},
+	}
+	assert.Equal(t, expected, dep.Spec)
+
+	// Prove idempotence
+	beforeErrCount = testutils.GetErrMetricCount(t, placeholderPodControllerName)
+	beforeReconcileCount = testutils.GetReconcileMetricCount(t, placeholderPodControllerName, metrics.LabelSuccess)
+	_, err = p.Reconcile(ctx, req)
+	require.NoError(t, err)
+	require.Equal(t, testutils.GetErrMetricCount(t, placeholderPodControllerName), beforeErrCount)
+	require.Greater(t, testutils.GetReconcileMetricCount(t, placeholderPodControllerName, metrics.LabelSuccess), beforeReconcileCount)
+
+	// Change the ingress resource's class
+	ing.Spec.IngressClassName = nil
+	require.NoError(t, c.Update(ctx, ing))
+
+	beforeErrCount = testutils.GetErrMetricCount(t, placeholderPodControllerName)
+	beforeReconcileCount = testutils.GetReconcileMetricCount(t, placeholderPodControllerName, metrics.LabelSuccess)
+	_, err = p.Reconcile(ctx, req)
+	require.NoError(t, err)
+	require.Equal(t, testutils.GetErrMetricCount(t, placeholderPodControllerName), beforeErrCount)
+	require.Greater(t, testutils.GetReconcileMetricCount(t, placeholderPodControllerName, metrics.LabelSuccess), beforeReconcileCount)
+
+	// Prove the deployment was not deleted
+	require.False(t, errors.IsNotFound(c.Get(ctx, client.ObjectKeyFromObject(dep), dep)))
+
+	// Prove idempotence
+	require.False(t, errors.IsNotFound(c.Get(ctx, client.ObjectKeyFromObject(dep), dep)))
```

Comment:
what is trying to prove idempotence mean in this case? we already proved it existed in 287